subsys/net/lib/openthread/platform/radio_spinel.cpp

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright (c) 2022, Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread platform abstraction
+ *   for radio communication using Spinel.
+ *
+ */
+#include <openthread-system.h>
+#include <openthread/platform/radio.h>
+#include <zephyr/fatal.h>
+#include <zephyr/sys/util_macro.h>
+
+#include <lib/spinel/radio_spinel.hpp>
+
+#ifdef CONFIG_OPENTHREAD_HOSTPROCESSOR_SPI
+#include "spi_interface.hpp"
+
+typedef ot::Spinel::RadioSpinel<SpiInterface, RadioSpinelContext> PlatformSpiRadioSpinel;
```

Comment:
```suggestion
typedef ot::Spinel::RadioSpinel<SpiInterface, RadioSpinelContext> PlatformRadioSpinel;
```

Reply:
1. Yes, kind of, we could try to re-implement the entire `Spinel` protocol, however this would be almost impossible to maintain.
2. Yes